### PR TITLE
Flashing Fix

### PR DESCRIPTION
**Pull request by ikeough**
This PR does the following:
- Invokes rendering on the UI thread with background priority to remove the "flashing" effect when resizing the helix viewport. This has the undesirable affect of rescaling the rendered image until such time as the Dispatcher can process the request to render. But it's better than flashing. Also, on some machines, when the view is resized slowly enough, the flashing would result in the view NEVER redrawing.

### NOTE: Please review #186 before this PR, as this PR is based on that.  

FYI @objorke

### CODE REVIEW

**Comment by objorke** [comment]
I would like to merge this enhancement, but the build is broken.. Can you see what is wrong?

File: Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj [hunk 1]
```diff
@@ -45,9 +46,11 @@
       <HintPath>..\..\..\packages\PropertyTools.Wpf.2014.1.29.1\lib\NET40\PropertyTools.Wpf.dll</HintPath>
     </Reference>
     <Reference Include="SharpDX">
-      <HintPath>..\..\..\packages\SharpDX.2.5.0\lib\net40\SharpDX.dll</HintPath>
+      <HintPath>$(SharpDXPackageBinDir)\SharpDX.dll</HintPath>
```
Comment (objorke) [inline Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj]: shouldn't it point to the nuget package?

File: Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj [hunk 2]
```diff
@@ -45,9 +46,11 @@
       <HintPath>..\..\..\packages\PropertyTools.Wpf.2014.1.29.1\lib\NET40\PropertyTools.Wpf.dll</HintPath>
     </Reference>
     <Reference Include="SharpDX">
-      <HintPath>..\..\..\packages\SharpDX.2.5.0\lib\net40\SharpDX.dll</HintPath>
+      <HintPath>$(SharpDXPackageBinDir)\SharpDX.dll</HintPath>
     </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Drawing" />
+    <Reference Include="System.Windows.Forms" />
```
Comment (objorke) [inline Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj]: Is this really needed in a WPF project?

File: Source/Examples/WPF.SharpDX/SimpleDemo/MainViewModel.cs [hunk 3]
```diff
@@ -49,7 +49,7 @@ public MainViewModel()
 
             // camera setup
             this.Camera = new PerspectiveCamera { Position = new Point3D(3, 3, 5), LookDirection = new Vector3D(-3, -3, -5), UpDirection = new Vector3D(0, 1, 0) };
-
+            ((PerspectiveCamera) this.Camera).FarPlaneDistance = 5000000;
```
Comment (objorke) [inline Source/Examples/WPF.SharpDX/SimpleDemo/MainViewModel.cs]: can't this be set on the line above?

File: Source/HelixToolkit.Wpf.SharpDX.Tests/Controls/CanvasMock.cs [hunk 4]
```diff
@@ -23,15 +23,15 @@ public CanvasMock()
         public bool IsShadowMapEnabled { get; private set; }
         public bool IsMSAAEnabled { get; private set; }
         public IRenderer Renderable { get; private set; }
+        public RenderTechnique RenderTechnique { get; private set; }
```
Comment (objorke) [inline Source/HelixToolkit.Wpf.SharpDX.Tests/Controls/CanvasMock.cs]: I like this kind of cleanup :-) thanks!

File: Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj (hunk 1 above)
Comment (ikeough) [inline Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj]: This is an environment variable set by SharpDX's nuget package install script. It adds a `.targets` reference to the project to resolve that path. It works locally, I'm still trying to figure out why it doesn't work on AppVeyor.

File: Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj (hunk 2 above)
Comment (ikeough) [inline Source/Examples/WPF.SharpDX/DeferredShadingDemo/DeferredShadingDemo_NET40.csproj]: These are also, annoyingly, added by the SharpDX nuget installer script. They are not required, and I'll remove them.

**Comment by objorke** [comment]
could you squash this one too?

**Comment by ikeough** [comment]
@objorke This shows as squashed locally, because it's branched from the UpdateSharpDX... branch. I think github might be showing the pre-squashed state. That is, I can't squash it any more.

**Pull request merged**